// qubeulator-components/collection/detail.js
"use strict";
var _ = require("lodash");

var React = require("react");
var Router = require("react-router-component");
var Locations = Router.Locations;
var Location = Router.Location;

var QuestorDetailProgression = require("../shared/questor-detail-progression");

module.exports = React.createClass({
  displayName: "CollectionDetail",
  getInitialState: function () {
    return {
      evolution: 1,
      level: 0
    };
  },
  render: function () {
    // TODO: Determine why decodeURI is necessary on reload but not a clicked link
    var questor = _.find(this.props.questors, { name: decodeURI(this.props.questor) });

    return (
      Locations({ contextual: true, ref: "router" },
        Location({
          path: "/",
          handler: QuestorDetailProgression,
          questor: questor,
          evolution: this.state.evolution,
          level: this.state.level,
          onUpdate: this.handleUpdate
        }),
        Location({
          path: "/:evolution/:level",
          handler: QuestorDetailProgression,
          questor: questor,
          onUpdate: this.handleUpdate
        })
      )
    );
  },
  handleUpdate: function (evolution, level) {
    this.setState({
      evolution: evolution,
      level: level
    });

    this.refs.router.navigate(["/", evolution, level].join("/"));
  },
});
