// qubeulator-components/collection/index.js
"use strict";
var React = require("react");
var Router = require("react-router-component");
var Locations = Router.Locations;
var Location = Router.Location;

var CollectionList = require("./list");
var CollectionDetail = require("./detail");

module.exports = React.createClass({
  getInitialState: function () {
    return {
      affinity: "all",
      rarity: "0"
    };
  },
  render: function () {
    return (
      Locations({ contextual: true, ref: "router" },
        Location({
          path: "/",
          handler: CollectionList,
          questors: this.props.questors,
          affinity: this.state.affinity,
          rarity: this.state.rarity,
          onUpdate: this.handleUpdate
        }),
        Location({
          path: "/detail/:questor",
          handler: CollectionDetail,
          questors: this.props.questors
        }),
        Location({
          path: "/:affinity/:rarity",
          handler: CollectionList,
          questors: this.props.questors,
          onUpdate: this.handleUpdate
        })
      )
    );
  },
  handleUpdate: function (affinity, rarity) {
    this.setState({
      affinity: affinity,
      rarity: rarity
    });

    this.refs.router.navigate([affinity, rarity].join("/"));
  },
});
