// qubeulator-components/collection/index.js
"use strict";
var React = require("react");
var Router = require("react-router-component");
var Locations = Router.Locations;
var Location = Router.Location;

var CollectionList = require("./list");
var CollectionDetail = require("./detail");

module.exports = React.createClass({
  displayName: "Collection",
  getInitialState: function () {
    return {
      affinity: "all",
      rarity: "0",
      origin: "all"
    };
  },
  render: function () {
    return (
      Locations({ contextual: true, ref: "router" },
        Location({
          path: "/",
          handler: CollectionList,
          questors: this.props.questors,
          affinity: this.state.affinity,
          rarity: this.state.rarity,
          origin: this.state.origin,
          onUpdate: this.handleUpdate
        }),
        Location({
          path: "/detail/:questor\/?*",
          handler: CollectionDetail,
          questors: this.props.questors
        }),
        Location({
          path: "/:affinity/:rarity/:origin",
          handler: CollectionList,
          questors: this.props.questors,
          onUpdate: this.handleUpdate
        })
      )
    );
  },
  handleUpdate: function (affinity, rarity, origin) {
    this.setState({
      affinity: affinity,
      rarity: rarity,
      origin: origin
    });

    // Build URL based off updated state, checking for a trailing slash
    var url = [
      (this.refs.router.state.prefix.substr(-1) === "/") ? "" : "/",
      affinity,
      rarity,
      origin
    ].join("/");

    this.refs.router.navigate(url);
  },
});
