// qubeulator-components/models/questor.js
"use strict";
var _ = require("lodash");

var State = require("ampersand-state");

var Questor = State.extend({
  props: {
    name: "string",
    rarity: "string",
    affinity: "string",
    origin: "string",
    description: "string",
    baseAttributes: "object",
    upgrade: "object",
    evolution: {
      type: "number",
      default: 1
    },
    level: {
      type: "number",
      default: 0,
      values: [0, 1, 2, 3, 4, 5]
    }
  },
  derived: {
    currentAttributes: {
      deps: ["baseAttributes", "evolution", "level"],
      fn: function () {
        var curAttributes = _.clone(this.baseAttributes);

        curAttributes.strength = this.baseAttributes.strength + this.upgrade.strength * this.level;
        curAttributes.health = this.baseAttributes.health + this.upgrade.health * this.level;
        curAttributes.speed = this.baseAttributes.speed + this.upgrade.speed * this.level;

        if (this.level) {
          curAttributes.perSecond = this.baseAttributes.perSecond + this.upgrade.perSecondProgression[this.level - 1];
        }

        return curAttributes;
      },
    }
  }
});

module.exports = Questor;
