// qubeulator-components/shared/questor-detail-progression.js
"use strict";
var React = require("react");

var RadioGroup = require("react-radio-group");

var QuestorDetail = require("../shared/questor-detail");

var Questor = require("../models/questor");

module.exports = React.createClass({
  displayName: "QuestorDetailProgression",
  render: function () {
    var questor = new Questor(this.props.questor);

    try {
      questor.set({
        evolution: +this.props.evolution,
        level: +this.props.level
      });
    } catch (e) {}

    return (
      React.DOM.div(null,
        RadioGroup({
          name: "evolution",
          value: this.props.evolution,
          ref: "evolutionGroup",
          onChange: this.handleChange
        },
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "1" }), "1"),
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "2" }), "2"),
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "3" }), "3"),
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "4" }), "4")
        ),
        RadioGroup({
          name: "level",
          value: this.props.level,
          ref: "levelGroup",
          onChange: this.handleChange
        },
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "0" }), "0"),
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "1" }), "1"),
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "2" }), "2"),
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "3" }), "3"),
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "4" }), "4"),
          React.DOM.label(null, React.DOM.input({ type: "radio", value: "5" }), "5")
        ),
        QuestorDetail({ questor: questor })
      )
    );
  },
  handleChange: function () {
    this.props.onUpdate(
      this.refs.evolutionGroup.getCheckedValue(),
      this.refs.levelGroup.getCheckedValue()
    );
  }
});
